x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmAuthenticateTests.java

```diff
@@ -170,8 +172,152 @@ public void testPkcJwkSetUrlNotFound() throws Exception {
         }
     }
 
+    /**
+     * Verify that a JWT realm successfully handles rotating JWKSets.
+     *
+     * @throws Exception Unexpected test failure
+     */
+    public void testJwkSetRotationHttps() throws Exception {
+        final JwtRealmsService jwtRealmsService = this.generateJwtRealmsService(this.createJwtRealmsSettingsBuilder());
+        final String principalClaimName = randomFrom(jwtRealmsService.getPrincipalClaimNames());
+
+        final List<Realm> allRealms = new ArrayList<>(); // authc and authz realms
+        final boolean createHttpsServer = true; // force issuer to create HTTPS server for its PKC JWKSet
+        final JwtIssuer jwtIssuer = this.createJwtIssuer(0, principalClaimName, 12, 1, 1, 1, createHttpsServer);
```

Comment:
```suggestion
        final int algsCount = randomIntBetween(1,12);
        final JwtIssuer jwtIssuer = this.createJwtIssuer(0, principalClaimName, algsCount, 1, 1, 1, createHttpsServer);
```

I think it is a good idea to randomize the number of algorithms, so this test covers different use cases for reloading when using HMAC-only, PKC-only, or both together.